app/actions/remote/command.ts

```diff
@@ -0,0 +1,203 @@
+//         const errorResponse = res.error as AppCallResponse;
+//         return createErrorMessage(errorResponse.error || intl.formatMessage({
+//             id: 'apps.error.unknown',
+//             defaultMessage: 'Unknown error.',
+//         }));
+//     }
+//     const callResp = res.data as AppCallResponse;
+//     switch (callResp.type) {
+//     case AppCallResponseTypes.OK:
+//         if (callResp.markdown) {
+//             dispatch(postEphemeralCallResponseForCommandArgs(callResp, callResp.markdown, args));
+//         }
+//         return {data: {}};
+//     case AppCallResponseTypes.FORM:
+//     case AppCallResponseTypes.NAVIGATE:
+//         return {data: {}};
+//     default:
+//         return createErrorMessage(intl.formatMessage({
+//             id: 'apps.error.responses.unknown_type',
+//             defaultMessage: 'App response type not supported. Response type: {type}.',
+//         }, {
+//             type: callResp.type,
+//         }));
+//     }
+// };
+
+const filterEmDashForCommand = (command: string): string => {
+    return command.replace(/\u2014/g, '--');
+};
+
+export const handleGotoLocation = async (serverUrl: string, intl: IntlShape, location: string) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const config = await queryConfig(operator.database);
+    const match = matchDeepLink(location, serverUrl, config.SiteURL);
+
+    if (match) {
+        switch (match.type) {
+            case DeepLinkTypes.CHANNEL: {
+                const data = match.data as DeepLinkChannel;
+                switchToChannelByName(data.serverUrl, data.channelName, data.teamName, DraftUtils.errorBadChannel, intl);
+                break;
+            }
+            case DeepLinkTypes.PERMALINK: {
+                const data = match.data as DeepLinkPermalink;
+                showPermalink(serverUrl, data.teamName, data.postId, intl);
+                break;
+            }
+            case DeepLinkTypes.DMCHANNEL: {
+                const data = match.data as DeepLinkDM;
+                if (!data.userName) {
+                    DraftUtils.errorBadUser(intl);
+                    return {data: false};
+                }
+
+                let serverDatabase = operator.database;
+                if (data.serverUrl !== serverUrl) {
+                    serverDatabase = DatabaseManager.serverDatabases[serverUrl]?.database;
+                    if (!serverDatabase) {
+                        return {error: `${serverUrl} database not found`};
+                    }
+                }
+                const user = (await queryUsersByUsername(serverDatabase, [data.userName]))?.[0];
+                if (!user) {
+                    DraftUtils.errorBadUser(intl);
+                    return {data: false};
+                }
+
+                makeDirectChannel(data.serverUrl, user.id);
```

Comment:
This checks if a direct channel with this user exist, if not create it, and navigate there. Not sure if the name is the best one, but it is the one on V1.

Reply:
feel free to rename